### PR TITLE
Fixes supply fan heat effect is incorrect in standard ratings calculation for two speed DX cooling coil

### LINKED ISSUE #9528: Supply fan heat effect is incorrect in standard ratings calculation for two speed DX cooling coil
Issue overview
--------------
Supply fan heat effect used to determine standard net cooling capacity for standard ratings calculation in CalcTwoSpeedDXCoilStandardRating function is missing supply air mass flow rate as a multiplier in equation below:

```
            FanHeatCorrection = state.dataLoopNodes->Node(FanOutletNode).Enthalpy - state.dataLoopNodes->Node(FanInletNode).Enthalpy;
            NetCoolingCapRated = state.dataDXCoils->DXCoil(DXCoilNum).RatedTotCap(1) * TotCapTempModFac * TotCapFlowModFac - FanHeatCorrection;
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, any)
 - Version of EnergyPlus (Develop, [f361c6a](https://github.com/NREL/EnergyPlus/commit/f361c6a4ac42b83f11ec832ce483465bbc2cf371))

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #9528
 - Corrects the fan heat effect used to calculate net cooling capacity, EER, IEER and COPs of Packaged VAV unit ratings per ANSI/AHRI Standard 340/360-2007. The standard rating metric values will slightly change depending on the supply air flow rate.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -15632,7 +15634,8 @@ Real64 CalcTwoSpeedDXCoilIEERResidual(EnergyPlusData &state,
                                         FanStaticPressureRise);
         }
 
-        FanHeatCorrection = state.dataLoopNodes->Node(FanOutletNodeNum).Enthalpy - state.dataLoopNodes->Node(FanInletNodeNum).Enthalpy;
+        FanHeatCorrection =
+            SupplyAirMassFlowRate * (state.dataLoopNodes->Node(FanOutletNodeNum).Enthalpy - state.dataLoopNodes->Node(FanInletNodeNum).Enthalpy);
```
Comment (Nigusse) [inline src/EnergyPlus/DXCoils.cc:15638]: Air enthalpy difference across the supply fan should have been multiplied by air mass flow rate.

File: tst/EnergyPlus/unit/DXCoils.unit.cc [hunk 2]
```diff
@@ -4330,4 +4331,134 @@ TEST_F(EnergyPlusFixture, MultiSpeedDXCoolingCoilOutputTest)
...
+    EXPECT_TRUE(coolcoilTwoSpeed.RateWithInternalStaticAndFanObject);
+    EXPECT_EQ("8.77", RetrievePreDefTableEntry(*state, state->dataOutRptPredefined->pdchDXCoolCoilEERIP, coolcoilTwoSpeed.Name));
+    EXPECT_EQ("11.25", RetrievePreDefTableEntry(*state, state->dataOutRptPredefined->pdchDXCoolCoilIEERIP, coolcoilTwoSpeed.Name));
+    // test 2: using default fan power per evap air flow rate, 365 W/1000 scfm or 773.3 W/(m3/s)
+    coolcoilTwoSpeed.RateWithInternalStaticAndFanObject = false;
+    OutputReportPredefined::SetPredefinedTables(*state);
+    CalcTwoSpeedDXCoilStandardRating(*state, dXCoilIndex);
+    EXPECT_EQ("8.72", RetrievePreDefTableEntry(*state, state->dataOutRptPredefined->pdchDXCoolCoilEERIP, coolcoilTwoSpeed.Name));
+    EXPECT_EQ("10.15", RetrievePreDefTableEntry(*state, state->dataOutRptPredefined->pdchDXCoolCoilIEERIP, coolcoilTwoSpeed.Name));
+}
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/DXCoils.unit.cc:4462]: Unit test verifies EER and IEER values calculation using the internal static pressure rise and default methods.

**Review by Nigusse — COMMENTED** [review]
Corrects the fan heat effect used to calculate net cooling capacity, EER, and IEER of Packaged VAV unit ratings per ANSI/AHRI Standard 340/360-2007.

**Comment by Nigusse** [comment]
There are 27 example files with `Coil:Cooling:DX:TwoSpeed` object. These example files may produce diffs if the supply fan heat effect is calculated using `Internal Static Air Pressure`, i.e., if the `RateWithInternalStaticAndFanObject` variable is set to true. Diffs are expected in `*.eio` and `*.htm` files only. In total there are 10 example files that has diffs in `*.eio` and `*.htm` files but 4 of these example files also show `ESO small diffs` and `MTR small diffs`.

**Comment by Nigusse** [comment]
Sample diffs based on `5ZoneGeometryTransform.idf` example file.  The standard rating metrics `Standard Net Cooling Capacity `, `IEER {Btu/W-h}`, `COP{W/W}`, and `IEER {Btu/W-h}` values were slightly reduced because the fan heat effect has increased due to this fix. Note that the supply flow rate is `> 1.0 m3/s`.

```
  Component Sizing Information, Fan:VariableVolume, SUPPLY FAN 1, Design Size Maximum Flow Rate [m3/s], 2.55511
 ! <VAV DX Cooling Coil Standard Rating Information>, DX Coil Type, DX Coil Name, Fan Type, Fan Name, Standard Net Cooling Capacity {W}, Standard Net Cooling Capacity {Btu/h}, IEER {Btu/W-h}, COP 100% Capacity {W/W}, COP 75% Capacity {W/W}, COP 50% Capacity {W/W}, COP 25% Capacity {W/W}, EER 100% Capacity {Btu/W-h}, EER 75% Capacity {Btu/W-h}, EER 50% Capacity {Btu/W-h}, EER 25% Capacity {Btu/W-h}, Supply Air Flow 100% {kg/s}, Supply Air Flow 75% {kg/s},Supply Air Flow 50% {kg/s},Supply Air Flow 25% {kg/s}
- VAV DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:TwoSpeed,MAIN COOLING COIL 1,Fan:VariableVolume,SUPPLY FAN 1,42207.57,144018.19,13.60,2.61,3.51,4.56,5.44,8.90,11.99,15.57,18.56,2.8859,2.2669,1.6093,0.7177,
+ VAV DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:TwoSpeed,MAIN COOLING COIL 1,Fan:VariableVolume,SUPPLY FAN 1,41014.54,139947.41,13.48,2.54,3.46,4.56,5.45,8.65,11.81,15.55,18.60,2.8859,2.2481,1.5845,0.6844,
```

**Comment by Nigusse** [comment]
Sample diffs for `_DOASDXCOIL_wUserSHRMethod.idf` example file. The `Standard Net Cooling Capacity` , `IEER {Btu/W-h}`, `COP{W/W}`, and `EER {Btu/W-h}` values were slightly increased because the fan heat effect has increased due to this fix. Note that in this example file the supply air flow rate value are `< 1.0 m3/s`, so as a multiplier reduces the fan heat effect (`=m_dot * deltaEnthalpy`) and increases the standard ratings metric values. 
```
 ! <VAV DX Cooling Coil Standard Rating Information>, DX Coil Type, DX Coil Name, Fan Type, Fan Name, Standard Net Cooling Capacity {W}, Standard Net Cooling Capacity {Btu/h}, IEER {Btu/W-h}, COP 100% Capacity {W/W}, COP 75% Capacity {W/W}, COP 50% Capacity {W/W}, COP 25% Capacity {W/W}, EER 100% Capacity {Btu/W-h}, EER 75% Capacity {Btu/W-h}, EER 50% Capacity {Btu/W-h}, EER 25% Capacity {Btu/W-h}, Supply Air Flow 100% {kg/s}, Supply Air Flow 75% {kg/s},Supply Air Flow 50% {kg/s},Supply Air Flow 25% {kg/s}
- VAV DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:TwoSpeed,DOAS COOLING COIL,Fan:VariableVolume,DOAS SUPPLY FAN,19747.56,67381.46,14.32,2.77,3.74,4.82,5.51,9.45,12.75,16.45,18.79,0.5671,0.4455,0.3186,0.1633,
+ VAV DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:TwoSpeed,DOAS COOLING COIL,Fan:VariableVolume,DOAS SUPPLY FAN,19995.48,68227.40,14.63,2.80,3.81,4.94,5.66,9.57,12.98,16.87,19.31,0.5671,0.4440,0.3162,0.1619,
```

**Comment by Nigusse** [comment]
Four of the 10 example files that has `*.eio` and tabular `*.htm` output files also has small diffs in `*.ESO` and `*.MTR` files. I was not expecting any small diffs in `*.ESO` and `*.MTR` files because this fix impacts the standard ratings calculation only. The only explanation I can think of is a secondary effect from the use of CalcMultiSpeedDXCoil() coil performance calculation function.

**Comment by Nigusse** [comment]
This PR is ready for review.

**Comment by Nigusse** [comment]
I ran locally one of the four example files `(RefMedOffVAVAllDefVRP.idf)` with small diffs in *.ESO and *.MTR files. See plots for a report variables with largest diffs for develop and the branch. The red color lines are the diffs.   

image

image

File: src/EnergyPlus/DXCoils.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/DXCoils.cc:15638]: Yep, this seems like a great change.

File: tst/EnergyPlus/unit/DXCoils.unit.cc (hunk 2 above)
Comment (Myoldmopar) [reply tst/EnergyPlus/unit/DXCoils.unit.cc:4462]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
This looks 100% ready.  CI is stable, although the linux release looks a bit funny with the GPU diff.  I would merge this right now, but it will end up causing diffs in all the other CI results I'm waiting on today.  I'll wait until later but I believe this is ready.

**Comment by Myoldmopar** [comment]
OK building and testing this locally.  Assuming it is still happy, this will be the first in a chunk of PRs to drop this afternoon.  They are some diff-causing PRs that are all ready to go and today is the day.

**Comment by Myoldmopar** [comment]
Pulled develop in and re-ran, everything is clean.  This is merging now.  Thanks @Nigusse

**Pull request merged**